### DISCUSSION TITLE
post install script doesn't work in proxmox 8.3.0

### DISCUSSION THREAD

**Discussion by svritc** [community-scripts/ProxmoxVE#485] [Q&A]
Start the Proxmox VE Post Install Script (y/n)?y
 ✗ This version of Proxmox Virtual Environment is not supported
Requires Proxmox Virtual Environment Version 8.0 or later.
Exiting...

**Comment by MickLesk** [comment]
Which Bash do you call? Its fixed since a few days 

Tested now, works fine

**Reply by Chuaser** [reply]
"disable nag subscription" does not work

**Reply by svritc** [reply]
Thank you so much its working now..

**Reply by MickLesk** [reply]
> "disable nag subscription" does not work

and what not working there?
I didnt see any changes related to 8.2 and 8.1.
Without Userinput i cant fix anything.

Ive installed Proxmox on a 2nd pc and its running fine too.

**Reply by Chuaser** [reply]
> > "disable nag subscription" does not work
> 
> and what not working there? I didnt see any changes related to 8.2 and 8.1. Without Userinput i cant fix anything.
> 
> Ive installed Proxmox on a 2nd pc and its running fine too.

script -> Proxmox VE Post Install
clean install 
 - pve iso 8.2.2, work fine script.
but 
clean install 
- pve 8.3 iso .. no working disabling the subscription nag.

Something strange is happening with 8.3 iso.

**Reply by MickLesk** [reply]
Can you do this please?

```bash
cat <<EOF >/tmp/nag_check.sh
#!/bin/bash

# Description: Checks if the subscription nag patch has been applied in the Proxmox UI.

FILE="/usr/share/javascript/proxmox-widget-toolkit/proxmoxlib.js"
SEARCH_PATTERN="data\\.status.*{"

echo "Checking if the file $FILE exists..."
if [ -z "$FILE" ]; then
    echo "Error: FILE variable is not set. Exiting."
    exit 1
fi

if [ ! -f "$FILE" ]; then
    echo "File $FILE not found! Exiting."
    exit 1
fi

echo "Searching for relevant entries in the file..."
MATCH=$(grep -n -E "$SEARCH_PATTERN" "$FILE" | head -n 1)
if [ -z "$MATCH" ]; then
    echo "No matching entries found. The file may have been modified."
    exit 1
fi

LINE_NUMBER=$(echo "$MATCH" | cut -d: -f1)
echo "Found relevant entry on line $LINE_NUMBER:"

# Display +/- 2 lines around the matched line
START=$((LINE_NUMBER - 2))
if [ "$START" -lt 1 ]; then START=1; fi
END=$((LINE_NUMBER + 2))
sed -n "${START},${END}p" "$FILE"

# Check if the patch is applied
if echo "$MATCH" | grep -q '!'; then
    echo "Patch NOT applied (contains '!')."
    exit 1
elif echo "$MATCH" | grep -q 'NoMoreNagging'; then
    echo "Patch successfully applied (contains 'NoMoreNagging')."
    exit 0
else
    echo "Unrecognized state. The file may have been altered."
    exit 1
fi
EOF
chmod +x /tmp/nag_check.sh
/tmp/nag_check.sh
```

**Reply by wotsthestory** [reply]
> > > "disable nag subscription" does not work
> > 
> > 
> > and what not working there? I didnt see any changes related to 8.2 and 8.1. Without Userinput i cant fix anything.
> > Ive installed Proxmox on a 2nd pc and its running fine too.
> 
> script -> Proxmox VE Post Install clean install
> 
>     * pve iso 8.2.2, work fine script.
>       but
>       clean install
> 
>     * pve 8.3 iso .. no working disabling the subscription nag.
> 
> 
> Something strange is happening with 8.3 iso.

I get exactly the same. Clean install with new 8.3 iso and error message above. Worked before on any version which has been upgraded up to and above.

**Reply by MickLesk** [reply]
Then Checkout my Script above and give feedback

**Reply by wotsthestory** [reply]
I actually discovered there was a difference between the two sites, I used the one from the community-scripts.... site and it worked perfectly. Sorry about that

**Reply by MickLesk** [reply]
ah okay ^^ you have used the tteck repo before?

**Reply by dedors** [reply]
I just run into the same issue with tteck script (as it is the first google search result).
community-scripts does work.
if someone lands here too: https://community-scripts.github.io/ProxmoxVE/scripts?id=post-pve-install

**Comment by svritc** [comment]
@MickLesk : Thanks for the script I have had tried your script and below is the results in my test server with fresh install:
root@pmx01:~# vi /tmp/nag_check.sh 
root@pmx01:~# chmod +x /tmp/nag_check.sh
root@pmx01:~# /tmp/nag_check.sh 
Checking if the file /usr/share/javascript/proxmox-widget-toolkit/proxmoxlib.js exists...
Searching for relevant entries in the file...
Found relevant entry on line 577:
                    let res = response.result;
                    if (res === null || res === undefined || !res || res
                        .data.status.toLowerCase() !== 'active') {
                        Ext.Msg.show({
                            title: gettext('No valid subscription'),
Patch NOT applied (contains '!').
root@pmx01:~#

**Comment by svritc** [comment]
for me I have tried the below script from another source and it is working in 8.3.0
curl --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/rickycodes/pve-no-subscription/main/no-subscription-warning.sh | sh

**Reply by MickLesk** [reply]
I can't reproduce this on any of my devices.
I have pulled a new Proxmox Iso, even there it works with my script.

Strangely enough, a script that is a few years old should work for you? 

--- 

I looked at the script, it does something completely different to ours. We remove the nag from the UI in “ /usr/share/javascript/proxmox-widget-toolkit/proxmoxlib.js” and the type replaces “NotFound” to “Active” in a file (/usr/share/perl5/PVE/API2/Subscription.pm)

I see absolutely no difference after running this. Then you are also the only one where it supposedly doesn't work.  :expressionless:

**Comment by JanWillemDeBoer** [comment]
Just installed 8.3.0 as well (clean install) and used [Proxmox VE Post Install ](https://community-scripts.github.io/ProxmoxVE/scripts?id=post-pve-install). After reboot I still have the nag too

**Reply by MickLesk** [reply]
cache cleaned? 

Where you see the nag? Screenshot? 
Have installed 4 times on different vms, never see nag after script ^^

**Reply by JanWillemDeBoer** [reply]
> cache cleaned?
> 
> Where you see the nag? Screenshot? Have installed 4 times on different vms, never see nag after script ^^

Yep, cache cleared, even used private mode, and even another browser.

<img width="448" alt="image" src="https://github.com/user-attachments/assets/f196da99-283e-4665-9aa3-133098f38bd5">

Clean install of 8.3.0 and used the bash script on [this](https://community-scripts.github.io/ProxmoxVE/scripts?id=post-pve-install) page. Even ran it multiple times now, all the other stuff seems to be updated, like repositories.

Also ran it before adding it to a cluster (and after it was added)

**Reply by MickLesk** [reply]
do you have entered all yes? 

Can you share an video?

**Reply by JanWillemDeBoer** [reply]
entered no on the HA, cause I use that.

I can make a video of selecting yes, but trust me, I've done this many times, and know the difference between yes and no :)

**Reply by MickLesk** [reply]
just want to understand it anyway, because I couldn't reproduce it on 3 PC's (Live) + at work + 4 VMs ^^

**Reply by JanWillemDeBoer** [reply]
Yeah, it's really weird, never had this before either. For me it's not really an issue, just fixed it myself in the code for now (not the script code, but the PVE files). Was just trying to help you improve your scripts +1.

**Comment by butterdori** [comment]
![250113_WindowsTerminal_2699](https://github.com/user-attachments/assets/ddc8aa79-6700-43bf-9f36-06bec2b48968)
I'm not sure if it's the same issue, but I don't get a prompt for the no-nag at all. Given that it asks me if I want to enable HA, this is a pattern that runs when the script has been run before? How can I revert and start fresh?

**Reply by MickLesk** [reply]
No nag Request come only If you answer all (ttecks Logic)

**Reply by butterdori** [reply]
![Screenshot From 2025-01-13 19-44-58](https://github.com/user-attachments/assets/8fd9ec75-da3c-48c8-88f9-e8568c12bfae)

On a different server, I answered all yes except for the pvetest repository, but it still showed prompt for no nag, so that seems strange. In any case, I tried again on the first machine and answered all yes, and still haven't got the prompt for no nag.

**Comment by PacketPalantir** [comment]
I had this happen yesterday on a brand new host. All I had to do was update the packages manually through the Proxmox GUI and then run the post-install script again.

**Comment by mgrimace** [comment]
Happening now on 8.4.5, re-ran the post-install script (since it's been awhile) and it the nag screen still appears on reboot.
Update: and hitting 'refresh' in PVE also says: `You do not have a valid subscription for this server. Please visit [www.proxmox.com](https://www.proxmox.com/en/proxmox-virtual-environment/pricing) to get a list of available options.`

**Reply by MickLesk** [reply]
rm /etc/apt/apt.conf.d/no-nag-script
And then Run the Script again

**Reply by carlopuri** [reply]
thanks a lot @MickLesk 
tried your solution on 8.4.5 (constantly updated from 8.2) where "no-nag screen" wasn't working; now it works
